lib/mirage_nat_lru.ml

```diff
@@ -67,47 +69,45 @@ module Storage = struct
       match mappings with
       | [] -> Lwt.return (Ok ())
       | m :: ms ->
-        let known (src, _dst) = L.LRU.mem src !t in
+        let known (src, _dst) = L.LRU.mem src t in
         let first_known = known m in
         if List.exists (fun x -> known x <> first_known) ms then Lwt.return (Error `Overlap)
         else (
           (* TODO: this is not quite right if all mappings already exist, because it's possible that
              the lookups are part of differing pairs -- this situation is pathological, but possible *)
-          t := List.fold_left (fun acc (a, b) -> L.LRU.add a (expiry, b) acc) !t mappings;
-          Lwt.return (Ok ())
+          mappings |> List.iter L.LRU.(fun (a, b) -> add a (expiry, b) t; trim t);
+          Lwt.return_ok ()
         )
 
     let delete t mappings =
       let t = L.table t in
-      t := List.fold_left (fun acc m -> L.LRU.remove m acc) !t mappings;
+      mappings |> List.iter (fun m -> L.LRU.remove m t);
       Lwt.return_unit
 
-    let pp f t =
-      let t = !t in
-      Fmt.pf f "%d/%d" (L.LRU.size t) (L.LRU.capacity t)
+    let pp f t = Fmt.pf f "%d/%d" (L.LRU.size t) (L.LRU.capacity t)
   end
 
   module TCP  = Subtable(struct module LRU = Port_cache let table t = t.tcp  type transport_channel = Mirage_nat.port * Mirage_nat.port end)
   module UDP  = Subtable(struct module LRU = Port_cache let table t = t.udp  type transport_channel = Mirage_nat.port * Mirage_nat.port end)
   module ICMP = Subtable(struct module LRU = Id_cache   let table t = t.icmp type transport_channel = Cstruct.uint16                    end)
 
   let reset t =
-    t.tcp := t.defaults.empty_tcp;
-    t.udp := t.defaults.empty_udp;
-    t.icmp := t.defaults.empty_icmp;
+    t.tcp <- t.defaults.empty_tcp ();
+    t.udp <- t.defaults.empty_udp ();
+    t.icmp <- t.defaults.empty_icmp ();
     Lwt.return ()
 
   let empty ~tcp_size ~udp_size ~icmp_size =
     let defaults = {
-      empty_tcp = Port_cache.empty tcp_size;
-      empty_udp = Port_cache.empty udp_size;
-      empty_icmp = Id_cache.empty icmp_size;
+      empty_tcp = (fun () -> Port_cache.create ~random:false tcp_size);
+      empty_udp = (fun () -> Port_cache.create ~random:false udp_size);
+      empty_icmp = (fun () -> Id_cache.create ~random:false icmp_size);
```

Comment:
`~random:false` can be switched on to jiggle the table. See [the doc](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Hashtbl.MakeSeeded.html) for more exciting features.